no such file or directory COMMIT_EDITMSG on new checkout

When checking out a repo, there is no `COMMIT_EDITMSG` file until a new commit is made. In the case of [elastic-apm-node](https://github.com/elastic/apm-agent-nodejs), we trigger the commit lint automatically, as part of the test process. This currently means a fresh check out can run the tests because @commitling/cli will crash due to being unable to find that file.

## Expected Behavior
There was no commit to have a failing message on, so it should not be a failure.

## Current Behavior
It crashes with this output:

<details>
  <summary>@commitlint/cli output</summary>
```sh
/home/stephenbelanger/Code/elastic-apm-node/node_modules/@commitlint/cli/lib/cli.js:113
	throw err;
	^

Error: ENOENT: no such file or directory, open '/home/stephenbelanger/Code/elastic-apm-node/.git/COMMIT_EDITMSG'
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! elastic-apm-node@2.5.1 lint-commit: `test/lint-commits.sh`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the elastic-apm-node@2.5.1 lint-commit script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/stephenbelanger/.npm/_logs/2019-03-03T19_30_57_446Z-debug.log
```
</details>


## Affected packages
* [x] cli
* [ ] core
* [ ] prompt
* [ ] config-angular

## Possible Solution
Consider it a "pass" if the file is not found. 

## Steps to Reproduce (for bugs)
1. Checkout _any_ repo for the first time, which uses @commitlint/cli
2. Trigger @commitlint/cli
3. ???
4. :cry: 

## Context
Currently this makes the contributing process painful, as new contributors will always run into this the first time they try to run our tests.

--- a/@commitlint/read/src/get-edit-commit.ts
+++ b/@commitlint/read/src/get-edit-commit.ts
@@ -15,7 +15,20 @@ export async function getEditCommit(
 	}
 
 	const editFilePath = await getEditFilePath(top, edit);
-	const editFile: Buffer = await fs.readFile(editFilePath);
+
+	let editFile: Buffer;
+	try {
+		editFile = await fs.readFile(editFilePath);
+	} catch (err) {
+		if ((err as NodeJS.ErrnoException).code === "ENOENT") {
+			throw new Error(
+				`No commit message file found at ${editFilePath}.\n` +
+					`--edit reads the message prepared by 'git commit' and is intended to run from a commit-msg hook. ` +
+					`If you want to lint existing history, use --from / --to instead; to lint a specific file, pass its path as --edit <file>.`,
+			);
+		}
+		throw err;
+	}
 
 	return [`${editFile.toString("utf-8")}\n`];
 }

--- a/@commitlint/read/src/get-edit-commit.ts
+++ b/@commitlint/read/src/get-edit-commit.ts
@@ -21,10 +21,13 @@ export async function getEditCommit(
 		editFile = await fs.readFile(editFilePath);
 	} catch (err) {
 		if ((err as NodeJS.ErrnoException).code === "ENOENT") {
+			const hint =
+				typeof edit === "string"
+					? `Check that the path passed to --edit exists and is readable.`
+					: `--edit reads the message prepared by 'git commit' and is intended to run from a commit-msg hook. If you want to lint existing history, use --from / --to instead; to lint a specific file, pass its path as --edit <file>.`;
 			throw new Error(
-				`No commit message file found at ${editFilePath}.\n` +
-					`--edit reads the message prepared by 'git commit' and is intended to run from a commit-msg hook. ` +
-					`If you want to lint existing history, use --from / --to instead; to lint a specific file, pass its path as --edit <file>.`,
+				`No commit message file found at ${editFilePath}. ${hint}`,
+				{ cause: err },
 			);
 		}
 		throw err;

--- a/@commitlint/read/src/read.test.ts
+++ b/@commitlint/read/src/read.test.ts
@@ -16,6 +16,14 @@ test("get edit commit message specified by the `edit` flag", async () => {
 	expect(actual).toEqual(expected);
 });
 
+test("throws an actionable error when --edit is used on a fresh repo without COMMIT_EDITMSG", async () => {
+	const cwd: string = await git.bootstrap();
+
+	await expect(read({ edit: true, cwd })).rejects.toThrow(
+		/No commit message file found.*commit-msg hook/s,
+	);
+});
+
 test("get edit commit message from git root", async () => {
 	const cwd: string = await git.bootstrap();
 

--- a/@commitlint/read/src/read.test.ts
+++ b/@commitlint/read/src/read.test.ts
@@ -24,6 +24,14 @@ test("throws an actionable error when --edit is used on a fresh repo without COM
 	);
 });
 
+test("throws an actionable error when --edit <file> points at a missing path", async () => {
+	const cwd: string = await git.bootstrap();
+
+	await expect(read({ edit: "does-not-exist", cwd })).rejects.toThrow(
+		/No commit message file found.*Check that the path passed to --edit/s,
+	);
+});
+
 test("get edit commit message from git root", async () => {
 	const cwd: string = await git.bootstrap();
 

